sdks/python/apache_beam/testing/test_utils.py

```diff
@@ -129,3 +131,47 @@ def delete_files(file_paths):
     raise RuntimeError('Clean up failed. Invalid file path: %s.' %
                        file_paths)
   FileSystems.delete(file_paths)
+
+
+def wait_for_subscriptions_created(subs, timeout=60):
+  """Wait for all PubSub subscriptions are created."""
+  return _wait_until_all_exist(subs, timeout)
+
+
+def wait_for_topics_created(topics, timeout=60):
+  """Wait for all PubSub topics are created."""
+  return _wait_until_all_exist(topics, timeout)
+
+
+def _wait_until_all_exist(components, timeout):
+  need_wait = set(components)
+  start_time = time.time()
+  while time.time() - start_time <= timeout:
+    for c in components:
```

Comment:
I think you can do `for c in needs_wait` instead to iterate over a shorter list and remove `c in need_wait` below.

Reply:
If I replace `components` with `needs_wait` in line 150, I'll get:

RuntimeError: Set changed size during iteration.

guess we cannot modify set during iteration.

Reply:
OK. Maybe add a TODO to clean this up later.